sbnana/CAFAna/Core/SpectrumLoader.cxx

```diff
@@ -328,6 +334,238 @@ namespace ana
+        // but aren't grouped with the other nominal histograms. Keep track of
+        // the results for nominals in these caches to speed those systs up.
+        CutVarCache<bool, TruthCut, caf::SRTrueInteractionProxy> nomTruthCutCache;
+        CutVarCache<double, TruthVar, caf::SRTrueInteractionProxy> nomTruthWeiCache;
+        CutVarCache<double, TruthVar, caf::SRTrueInteractionProxy> nomTruthVarCache;
+
+        for(auto& shiftdef: spillcutdef.second){
+          const SystShifts& shift = shiftdef.first;
+
+          // Need to provide a clean slate for each new set of systematic
+          // shifts to work from. Copying the whole StandardRecord is pretty
+          // expensive, so modify it in place and revert it afterwards.
+
+          caf::SRProxySystController::BeginTransaction();
+
+          bool shifted = false;
+
+          double systWeight = 1;
+          // Can special-case nominal to not pay cost of Shift()
+          if(!shift.IsNominal()){
+            shift.Shift(&nu, systWeight);
+            // If there were only weighting systs applied then the cached
+            // nominal values are still valid.
+            shifted = caf::SRProxySystController::AnyShifted();
+          }
+
+          for(auto& truthcutdef: shiftdef.second){
+
+            const TruthCut& truthcut = truthcutdef.first;
+
+            const bool truthpass = shifted ? truthcut(&nu) : nomTruthCutCache.Get(truthcut, &nu);
+
+            // TruthCut failed, skip all the histograms that depended on it
+            if(!truthpass) continue;
+
+            for(auto& truthweidef: truthcutdef.second){
+
+              const TruthVar& truthweivar = truthweidef.first;
+
+              double truthwei = shifted ? truthweivar(&nu) : nomTruthWeiCache.Get(truthweivar, &nu);
+
+              truthwei *= systWeight;
+              if(truthwei == 0) continue;
+
+              for(auto& truthvardef: truthweidef.second){
+
+                // if TruthMultiVar
+                if(truthvardef.first.IsMulti()){
+                  for(double truthval: truthvardef.first.GetMultiVar()(&nu)){
+                    for(Spectrum* s: truthvardef.second.spects)
+                      s->Fill(truthval, truthwei);
+                  }
+                }
+                // if TruthVar
+                else{
+
+                  const TruthVar& truthvar = truthvardef.first.GetVar();
+                  const double truthval = shifted ? truthvar(&nu) : nomTruthVarCache.Get(truthvar, &nu);
+
+                  if(std::isnan(truthval) || std::isinf(truthval)){
+                    std::cerr << "Warning: Bad value: " << truthval
+                              << " returned from a TruthVar. The input variable(s) could "
+                              << "be NaN in the CAF, or perhaps your "
+                              << "Var code computed 0/0?";
+                    std::cout << " Not filling into this histogram for this slice." << std::endl;
```

Comment:
Should say `Not filling into this histogram for this mc.nu` or `Not filling into this histogram for this True Interaction` or etc.

Reply:
L414 updated to
```
                    std::cout << " Not filling into this histogram for this True Interaction." << std::endl;
```

Reply:
Great!